Tests/xSharePoint/xSharePoint.xSPUserProfileSection.Tests.ps1

```diff
@@ -0,0 +1,276 @@
+        }
+        Import-Module $Global:CurrentSharePointStubModule -WarningAction SilentlyContinue  
+        
+        
+        try { [Microsoft.Office.Server.UserProfiles] }
+        catch {
+            Add-Type @"
+                namespace Microsoft.Office.Server.UserProfiles {
+                public enum ConnectionType { ActiveDirectory, BusinessDataCatalog };
+                public enum ProfileType { User};
+                }        
+"@ -ErrorAction SilentlyContinue
+        }   
+
+        Import-Module (Join-Path ((Resolve-Path $PSScriptRoot\..\..).Path) "Modules\xSharePoint")
+        
+        $coreProperty = @{ 
+                            DisplayName = $testParams.DisplayName
+                            Name = $testParams.Name
+                        } | Add-Member ScriptMethod Commit {
+                            $Global:xSPUPSPropertyCommitCalled = $true
+                        } -PassThru | Add-Member ScriptMethod Delete {
+                            $Global:xSPUPSPropertyDeleteCalled = $true
+                        } -PassThru
+        $subTypeProperty = @{
+                            Name= $testParams.Name
+                            DisplayName= $testParams.DisplayName
+                            DisplayOrder =$testParams.DisplayOrder
+                            CoreProperty = $coreProperty
+                        }| Add-Member ScriptMethod Commit {
+                            $Global:xSPUPSPropertyCommitCalled = $true
+                        } -PassThru
+        $userProfileSubTypePropertiesNoProperty = @{} | Add-Member ScriptMethod Create {
+        param($section)
+                            $Global:xSPUPSubTypeCreateCalled = $true
+                        } -PassThru  | Add-Member ScriptMethod GetSectionByName {
+                            $result = $null
+                            if($Global:xSPUPGetSectionByNameCalled -eq $TRUE){
+                                $result = $subTypeProperty
+                            }
+                            $Global:xSPUPGetSectionByNameCalled  = $true
+                            return $result
+                        } -PassThru| Add-Member ScriptMethod Add {
+                            $Global:xSPUPSubTypeAddCalled = $true
+                        } -PassThru -Force 
+        $coreProperties = @{ProfileInformation = $coreProperty}
+        $userProfileSubTypePropertiesProperty = @{"ProfileInformation" = $subTypeProperty } | Add-Member ScriptMethod Create {
+                            $Global:xSPUPSubTypeCreateCalled = $true
+                        } -PassThru | Add-Member ScriptMethod Add {
+                            $Global:xSPUPSubTypeAddCalled = $true
+                        } -PassThru -Force <#| Add-Member ScriptMethod GetSectionByName {
+                            $Global:xSPUPGetSectionByNameCalled  = $true
+                            return $subTypeProperty
+                        } -PassThru
+                        #>
```

Comment:
Yeah, they should all be gone. I though I had removed them. Will check it out when I get home. 


Camilo Borges
Sent via iPhone while out and about


> On 17 Feb 2016, at 5:15 PM, Brian Farnhill <notifications@github.com> wrote:
> 
> In Tests/xSharePoint/xSharePoint.xSPUserProfileSection.Tests.ps1:
> 
> > +                            }
> > +                            $Global:xSPUPGetSectionByNameCalled  = $true
> > +                            return $result
> > +                        } -PassThru| Add-Member ScriptMethod Add {
> > +                            $Global:xSPUPSubTypeAddCalled = $true
> > +                        } -PassThru -Force 
> > +        $coreProperties = @{ProfileInformation = $coreProperty}
> > +        $userProfileSubTypePropertiesProperty = @{"ProfileInformation" = $subTypeProperty } | Add-Member ScriptMethod Create {
> > +                            $Global:xSPUPSubTypeCreateCalled = $true
> > +                        } -PassThru | Add-Member ScriptMethod Add {
> > +                            $Global:xSPUPSubTypeAddCalled = $true
> > +                        } -PassThru -Force <#| Add-Member ScriptMethod GetSectionByName {
> > +                            $Global:xSPUPGetSectionByNameCalled  = $true
> > +                            return $subTypeProperty
> > +                        } -PassThru
> > +                        #>
> More comments here - I take it this isn't actually needed?
> 
> —
> Reply to this email directly or view it on GitHub.
>